### PR TITLE
Rename global variables naming style and add debug level

### PR DESCRIPTION
**Pull request by AllanZyne**
- Rename exported global variables to spirv style
- add spirv_AsanDebugLevel variable to control printing debug message in libdevice, and reuse "UR_LOG_SANITIZER" to control status (like ur logger)

### CODE REVIEW

File: source/loader/layers/sanitizer/asan_interceptor.cpp [hunk 1]
```diff
@@ -25,14 +25,15 @@ constexpr int kUsmHostRedzoneMagic = (char)0x82;
 constexpr int kUsmSharedRedzoneMagic = (char)0x83;
 constexpr int kMemBufferRedzoneMagic = (char)0x84;
 
-const auto kSPIR_AsanShadowMemoryGlobalStart = "__AsanShadowMemoryGlobalStart";
-const auto kSPIR_AsanShadowMemoryGlobalEnd = "__AsanShadowMemoryGlobalEnd";
-const auto kSPIR_AsanShadowMemoryLocalStart = "__AsanShadowMemoryLocalStart";
-const auto kSPIR_AsanShadowMemoryLocalEnd = "__AsanShadowMemoryLocalEnd";
+const auto kSPIR_ShadowGlobalStart = "__spirv_AsanShadowGlobalStart";
```
Comment (wenju-he) [inline source/loader/layers/sanitizer/asan_interceptor.cpp:28]: what is the rational of using __spirv prefix?

Comment (AllanZyne) [reply source/loader/layers/sanitizer/asan_interceptor.cpp:28]: We instrument on SPIRV code, so I think using the SPIRV naming style would make the code look consistent.

Comment (wenju-he) [reply source/loader/layers/sanitizer/asan_interceptor.cpp:28]: SPIRV can have any functions, including functions that don't starts with __spirv

Comment (wenju-he) [reply source/loader/layers/sanitizer/asan_interceptor.cpp:28]: LGTM

**Review by wenju-he — APPROVED** [review]
LGTM

**Review by zhaomaosu — APPROVED** [review]
lgtm

**Pull request closed without merging**